src/basic/virt.c

```diff
@@ -471,6 +470,8 @@ int detect_container(void) {
                 /* If we are PID 1 we can just check our own environment variable, and that's authoritative. */
 
                 e = getenv("container");
+                if (!e)
+                        goto check_sched;
                 if (isempty(e)) {
```

Comment:
the isempty() check kinda does the same thing, more or less... we should keep only one

Reply:
It's on purpose, I wanted to treat the lack of the variable different than an empty assignment. I.e. undefined → do other checks, empty → no container, non-empty → container. WDYT?